### DISCUSSION TITLE
How to protect PocketBase from DDoS?

### DISCUSSION THREAD

**Discussion by ilshm** [pocketbase/pocketbase#4363] [Q&A]
**Hello! I'd like to know if there's a way to secure PocketBase in a production environment against automated attacks, potentially by integrating it with CloudFlare.**

**Comment by erickythierry** [comment]
I use cloudflare tunnel to expose the address of my pocketbase instance, so all communication between pocketbase and clients goes through the cloudflare network, so cloudflare protects the real IP of my server and applies cloudflare's protections against ddos.

However, this has a catch: delay
Using this solution I noticed a delay of a few tens of ms between my frontend and pocketbase. However, for me it is not a problem, since this solution gives me more security in exposing my instance.

[cloudflare tunnel](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/)

**Reply by ilshm** [reply]
Thanks, will try!
